openapi_python_client/utils.py

```diff
@@ -25,22 +32,23 @@ def fix_reserved_words(value: str) -> str:
     return value
 
 
-def group_title(value: str) -> str:
-    value = re.sub(r"([A-Z]{2,})([A-Z][a-z]|[ \-_]|$)", lambda m: m.group(1).title() + m.group(2), value.strip())
-    value = re.sub(r"(^|[ _-])([A-Z])", lambda m: m.group(1) + m.group(2).lower(), value)
-    return value
-
-
 def snake_case(value: str) -> str:
-    return fix_keywords(stringcase.snakecase(group_title(sanitize(value))))
+    words = split_words(sanitize(value))
+    value = "_".join(words).lower()
+    return fix_keywords(value)
 
 
 def pascal_case(value: str) -> str:
-    return fix_keywords(stringcase.pascalcase(sanitize(value.replace(" ", ""))))
+    words = split_words(sanitize(value))
+    words = [word.capitalize() if not word.isupper() else word for word in words]
```

Comment:
This can probably be a generator instead of list comprehension to save a bit of memory

Reply:
Yep. Looks like you fixed it already. :)